dom/src/DOMSource.ts

```diff
@@ -1,13 +1,23 @@
 import {Stream, MemoryStream} from 'xstream';
 import {PreventDefaultOpt} from './fromEvent';
+import {MainDOMSource} from './MainDOMSource';
+import {DocumentDOMSource} from './DocumentDOMSource';
+import {BodyDOMSource} from './BodyDOMSource';
+import {MockedDOMSource} from './mockDOMSource';
 export interface EventsFnOptions {
   useCapture?: boolean;
   preventDefault?: PreventDefaultOpt;
 }
 
-export interface DOMSource {
+export type DOMSource =
+  | MainDOMSource
+  | DocumentDOMSource
+  | BodyDOMSource
+  | MockedDOMSource;
+
+export interface BaseDOMSource {
   select(selector: string): DOMSource;
-  elements(): MemoryStream<Document | Element | Array<Element> | string>;
+  elements(): MemoryStream<Document | HTMLBodyElement | Array<Element>>;
```

Comment:
I'm afraid the removal of the `string` case would break `cycle/html` which has `HTMLSource implements DOMSource` and `elements(): MemoryStream<string>`

Reply:
I see

Reply:
readded string, did not interfer with type inference